x-pack/legacy/plugins/ml/server/models/filter/filter_manager.ts

```diff
@@ -5,22 +5,75 @@
  */
 
 import Boom from 'boom';
+import { IScopedClusterClient } from 'src/core/server';
+
+import { DetectorRule, DetectorRuleScope } from '../../../common/types/detector_rules';
+
+export interface Filter {
+  filter_id: string;
+  description?: string;
+  items: string[];
+}
+
+export interface FormFilter {
+  filterId: string;
+  description?: string;
+  addItems?: string[];
+  removeItems?: string[];
+}
+
+export interface FilterRequest {
+  filter_id: string;
+  description?: string;
+  add_items?: string[];
+  remove_items?: string[];
+}
+
+interface FilterUsage {
+  jobs: string[];
+  detectors: string[];
+}
+
+interface FilterStats {
+  filter_id: string;
+  description?: string;
+  item_count: number;
+  used_by: FilterUsage;
+}
+
+interface FiltersInUse {
+  [id: string]: FilterUsage;
+}
+
+interface PartialDetector {
+  detector_description: string;
+  custom_rules: DetectorRule[];
+}
+
+interface PartialJob {
+  job_id: string;
+  analysis_config: {
+    detectors: PartialDetector[];
+  };
+}
 
 export class FilterManager {
-  constructor(callWithRequest) {
-    this.callWithRequest = callWithRequest;
+  private _client: IScopedClusterClient['callAsCurrentUser'];
+
+  constructor(client: any) {
```

Comment:
could be `IScopedClusterClient['callAsCurrentUser']` instead of `any`

Reply:
Done in https://github.com/elastic/kibana/pull/57597/commits/0e08a2349214889364434dce9a2969bcbd56680d